### PR TITLE
MINOR: fixed missing close of Iterator, used try-with-resource where appropriate

### PR DESCRIPTION
**Pull request by pkleindl**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java [hunk 1]
```diff
@@ -942,21 +943,20 @@ public void close() {}
...
 
-        {
-            final TopologyTestDriver testDriver = new TopologyTestDriver(topology, config);
+
+        try (final TopologyTestDriver testDriver = new TopologyTestDriver(topology, config)) {
             Assert.assertNull(
-                "Closing the prior test driver should have cleaned up this store and value.",
-                testDriver.getKeyValueStore("storeProcessorStore").get("a")
+                    "Closing the prior test driver should have cleaned up this store and value.",
+                    testDriver.getKeyValueStore("storeProcessorStore").get("a")
```
Comment (mjsax) [inline streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java]: Nit: avoid unnecessary reformatting. Also, we use 4 space indention, not 8 space.

**Review by mjsax — APPROVED** [review]
LGTM.

Call for second review @guozhangwang @bbejeck @vvcephei @ableegoldman @cadonna 

This PR is extracted from another PR against `trunk` to back port this improvement to older branches, too.

**Review by guozhangwang — APPROVED** [review]
LGTM!

File: streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java (hunk 1 above)
Comment (pkleindl) [reply streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java]: Fixed it, my IDE was set to 8 spaces for "continuation indent".
Is there any official style guide or export of accepted settings?

Comment (mjsax) [reply streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java]: > Is there any official style guide or export of accepted settings?

Not really :(

Rule of thumb is, avoid reformatting if possible :)

**Comment by mjsax** [comment]
Java8 passed. Java11 failure unrelated. Tracked via Jira tickets.

Retest this please.

File: streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java (hunk 1 above)
Comment (ableegoldman) [reply streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java]: @mjsax Should we include indentation settings here: http://kafka.apache.org/coding-guide.html ? It's mentioned in the scala guidelines...

Comment (mjsax) [reply streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java]: Feel free to do a PR against https://github.com/apache/kafka-site

**Comment by mjsax** [comment]
Merged to `trunk` and cherry picked to `2.2`, `2.1`, `2.0`, and `1.1`.

Thanks for the PR @pkleindl!

**Pull request merged**